articles/modules/ROOT/pages/an-overview-of-the-system-database.adoc

```diff
@@ -0,0 +1,59 @@
+= An overview of the system database
+:slug: an-overview-of-the-system-database
+:author: Vivek Saran
+:neo4j-versions: 4.0, 4.1
+:tags: system
+:category: operations
+
+Neo4j 4.0 and higher versions support the management of multiple databases within the same DBMS. All these databases are controlled through a special database called the `system` database.
+
+This article provides a brief architectural overview of the `system` database.
+
+The role of the `system` database is to **define the configuration** for the other databases. There could be various types of configuration for the databases. For example:
+
+* operational configuration
+** existence (exists or not)
+** status (online/offline)
+* security configuration (RBAC)
+* `neo4j.conf` (this is not yet maintained by the `system` database, but there are plans in the future versions)
+
+
+
+image:https://s3.amazonaws.com/dev.assets.neo4j.com/kb-content/configuration.gif[]
+
+
+A lot of interesting operational work happens (behind the scenes) both in standalone and clustered environments, which is handled by a component called the *reconciler*. 
+Refer to Fig 2 below.
+
+image:https://s3.amazonaws.com/dev.assets.neo4j.com/kb-content/reconciler.gif[]
+
+The `system` database is replicated across the cluster, and there is a leader for it just like any other database. Every database in 4.0 and going forward is in an independent raft group. That means core-1 could be the leader for the `system` database, whereas core-2 could be the leader for `mydb` database.
+Everything inside the `system` database is also stored in a graph data model. However, only DDL commands (CREATE, DROP, SHOW, etc.) could be executed in this database, not the regular Cypher commands (MATCH, for example). There are nodes representing other databases (`neo4j` and `mydb` in this example from Fig 2).
+
+The **reconciler** talks to another component called the **database manager**, which manages the actual databases. Every instance reconciles the local copies of all the databases. If the reconciler on core-2 (in the above example) knows that `mydb` became online in the leader (core-1), it ensures that `mydb` becomes online in this instance as well. The `system` database in every instance has a reconciler and a database manager.
+
+The reconciler’s job is to reconcile between the desired state (STOPPED, STARTED, DROPPED) and the current state.
```

Comment:
One important aspect of the reconciler is that it operates completely asynchronously and always from the replicated "source of truth" desired state as written into the system database. If a server for example is partitioned for a while from the rest of the cluster, or is falling behind for some reason, then it will not get the updates to the "desired state" and hence the reconciler on that server will not yet perform the operational changes that the other servers already have done. As soon as the cluster is back to normal though, with connectivity and all, then the reconciler will continue of course.